docs/psf_spec/high_level_overview.rst

```diff
@@ -0,0 +1,244 @@
+`~photutils.psf.BasicPSFPhotometry`, this block is the final step of a single iteration. While
+a maximum number of iterations can be specified for, e.g., 
+`~photutils.psf.IterativelySubtractedPSFPhotometry`, this block -- specifically, the ender half
+-- assesses whether all sources have been found within the ``image``. If no new sources have
+been found within an iteration, fitting can be stopped prematurely, avoiding wasteful computational
+time. Similarly, the culler aspect of this block examines sources found within an iteration for
+quality; it should calculate some goodness-of-fit criterion and reject sources picked up by the
+``Finder`` but fall below a given threshold. This block therefore accepts the ``stars``
+output `~astropy.table.Table` and returns a new `~astropy.table.Table` with low quality sources
+-- such as cosmic ray hits present in the image -- removed as well as a boolean flag indicating
+whether the iterative fitting process has reached a conclusion and can be terminated. Please refer
+to the :doc:`documentation <culler_and_ender>` for further details.
+
+
+.. _Single Object Model:
+
+Single Object Model
+^^^^^^^^^^^^^^^^^^^
+An additional block sitting between the `PSF model`_ and the Fitter_, the ``Single Object Model``
+extends the fitting capability of the class from purely point-source objects to extended sources.
+This block, effectively, convolves the PSF of the given observation with a physical light
+distribution, producing the flux seen in each pixel by the system of such an extended source. This
+block therefore requires as its inputs the PSF model and the object type -- a string representing
+the type of source (point-like, spiral galaxy, etc.) to create the intrinsic light distribution
+for -- and returns a new `~astropy.modeling.Fittable2DModel` instance, the convolution of the PSF
+and the true source. Further specifics on this block, including how it will link to `Scene Maker`_
+can be found in the :doc:`documentation <single_object_model>`.
+
+
+.. _Scene Maker:
+
+Scene Maker
+^^^^^^^^^^^
+The final block within the scope of PSF fitting, the scene maker is also the last to be implemented.
+Currently no PSF fitting class extends its functionality to include this block, but it will
+eventually generalize and possibly replace the `Group Maker`_ as the method of merging and assigning
+non-independent detections within a given image. While current grouping is simply the determining
+of point sources with potentially overlapping flux, the *scene* maker will allow for the grouping
+of both point sources into an overlapping, simultaneous fit group but also into a single extended
+object. Thus each iteration will require a step in which multiple-point-source extended objects are
+evaluated for their separation, and newly detected point sources are evaluated for potential
+assignment as part of an extended source. The details of this block are available :doc:`here
```

Comment:
This may be better discussed in #756 ... but this discussion makes me realize that maybe the scene maker and single object model should be one thing?  Or even if separate classes, maybe they should be described together in one document?

Reply:
They're basically "point + extended" sources extensions of group maker and the PSF model, so while linked slightly more obviously by the need to share a common ``object_type`` list and know what to do with each available choice in each instance, they're no more or less inter-connected than group maker and PSF model; it's just obvious what to do with those as the only potential object type is a point source, which is simple to handle.

We can discuss this further in the corresponding PR to get into the details, and, again, I'll make any changes decided upon in there in this branch too.